### ISSUE TITLE
[FEATURE] Add possibility to change HTML tag in SfButton component

### ISSUE THREAD

**Issue by qiqqq** [DivanteLtd/storefront-ui#1241]
**Describe how this feature will help us improve**
There are cases when an element that looks visually like a button, is just a link (it redirects to some location). From accessibility reasons, it should be `<a>` tag in HTML. Currently, there is not possible to have `<a>` tag which looks like a button. We should have a possibility to pass prop with HTML tag to SfButton component.

**Describe the solution you'd considered**
1) Add prop to pass HTML tag to SfButton component (with default value `button`)
2) Change `<button>` tag in SfButton component to `<component :is="htmlTag">`
3) Add appropriate styles if needed

**Comment by AdamPawlinski** [comment]
I prepared it slightly different using boolean prop you can change component from native button to SfLink component and there is another prop with link to use in "a" tag. Wdyt @qiqqq ?
